src/GNDStk/Node/src/compare.hpp

```diff
@@ -0,0 +1,80 @@
+
+// -----------------------------------------------------------------------------
+// operator==
+// -----------------------------------------------------------------------------
+
+template<
+   template<class...> class METADATA_CONTAINER_TWO,
+   template<class...> class CHILDREN_CONTAINER_TWO
+>
+bool operator==(
+   const Node<METADATA_CONTAINER_TWO,CHILDREN_CONTAINER_TWO> &two
+) const {
+   const auto &one = *this;
+
+   // same node?
+   // void* because template arguments could differ
+   if ((void*)&one == (void*)&two)
+      return true;
+
+   // ------------------------
+   // Fast/easy checks
+   // ------------------------
+
+   // name
+   if (one.name != two.name)
+      return false;
+
+   // sizes
+   if (one.metadata.size() != two.metadata.size()) return false;
+   if (one.children.size() != two.children.size()) return false;
+
+   // metadata
+   if (std::set<metaPair>(one.metadata.begin(), one.metadata.end()) !=
+       std::set<metaPair>(two.metadata.begin(), two.metadata.end()))
+      return false;
+
+   // ------------------------
+   // OK, not fast/easy :-(
```

Comment:
Nah, it'll work ;-)

I don't think users will be using this feature anyway. It's more for testing purposes.

Reply:
Yes, that's the thought, and is why I decided not to spend too much time trying to optimize that particular capability.